model/onboarding/onboarding.go

```diff
@@ -15,40 +15,25 @@
 // specific language governing permissions and limitations
 // under the License.
 
-package beater
+package onboarding
 
 import (
-	"context"
 	"time"
 
 	"github.com/elastic/beats/libbeat/beat"
 	"github.com/elastic/beats/libbeat/common"
-	"github.com/elastic/beats/libbeat/logp"
-
-	"github.com/elastic/apm-server/beater/config"
-	logs "github.com/elastic/apm-server/log"
-	"github.com/elastic/apm-server/publish"
-	"github.com/elastic/apm-server/transform"
 )
 
-func notifyListening(ctx context.Context, config *config.Config, reporter publish.Reporter) {
-	logp.NewLogger(logs.Onboarding).Info("Publishing onboarding document")
-	reporter(ctx, publish.PendingReq{
-		Transformables: []transform.Transformable{onboardingDoc{listenAddr: config.Host}},
-		Tcontext:       &transform.Context{RequestTime: time.Now()},
-	})
-}
-
-type onboardingDoc struct {
-	listenAddr string
+type Doc struct {
+	ListenAddr string
 }
 
-func (o onboardingDoc) Transform(tctx *transform.Context) []beat.Event {
+func (d Doc) Transform() []beat.Event {
 	return []beat.Event{{
-		Timestamp: tctx.RequestTime,
+		Timestamp: time.Now(),
```

Comment:
Theoretically there could be a larger gap in pushing an event into the queue for transformation and actually transforming events. We should aim for aligning on the `Timestamp` meaning for the events (some are now moved to the decode phase, whereas for the onboarding it is moved to the transform phase). Just recently there was a pipeline processor introduced adding the ingestion timestamp, so I think settling for the request timestamp (as you have done for the other events) makes more sense. Then I would suggest to change this back to also reflect the scheduling timestamp.